content/docs/rfcs/41/README.md

```diff
@@ -0,0 +1,85 @@
+
+# Wire Format Specification / Syntax
+
+### Constructor
+
+* `constructor(uint256 membershipDeposit, uint256 depth, address _poseidonHasher) public`
+    * `membershipDeposit` sets the amount of ETH that must be deposited to `register` to the RLN group.
+    * `depth` is the depth of the merkle tree that determines the maximum number of members to the RLN group.
+    * `_poseidonHasher` is the address of the poseidon hasher contract used in the `hash` function.
+
+### Methods
+
+* register: This function registers the `pubkey` to the list of members.
+For successful registration, ETH equal to the amount specified in `membershipDeposit` MUST be paid while calling this function.
+This function MUST fire the `MemberRegistered` event.
+    * `function register(uint256 pubkey) external payable`
+
+
+
+* registerBatch: This function registers multiple `pubkeys` to the list of members.
+For succesful registration, ETH equal to the amount specified in `membershipDeposit * pubkeys.length` MUST be paid while calling this function.
+This function MUST fire the `MemberRegistered` event for each registered member.
+    * `function registerBatch(uint256[] calldata pubkeys) external payable`
+
+
+* withdraw: This function accepts the `secret` of the public key at index`_pubkeyIndex` to withdraw the ETH equal to the`membershipDeposit` to the `receiver` address.
+This function MUST remove the associated member from the list of members.
+This function MUST fire the `MemberWithdrawn` event.
+    * `function withdraw(uint256 secret, uint256 _pubkeyIndex, address payable receiver) external`
+
+
+* withdrawBatch: This function accepts multiple `secrets` of the public keys at the indices `pubkeyIndexes` to withdraw the ETH equal to the `membershipDeposit * secrets.length` to the `receiver` address.
+This function MUST remove the associated members from the list of members.
+This function MUST fire the `MemberWithdrawn` event for each member slashed from the group.
+    * `function withdrawBatch(uint256[] calldata secrets, uint256[] calldata pubkeyIndexes, address payable[] calldata receivers) external`
+
+
+### Events
+
+* MemberRegistered: MUST trigger when a member is registered to the RLN group.
+    * `event MemberRegistered(uint256 indexed pubkey, uint256 indexed index)`
+
+
+* MemberWithdrawn: MUST  trigger when a member is slashed from the RLN group.
+    * `event MemberWithdrawn(uint256 indexed pubkey, uint256 indexed index)`
```

Comment:
It is true that in the current implementation, we keep the Merkle tree and emitting the pk may not be necessary, but consider the case that instead of replacing the leaf, a blacklist of Pks is maintained. Emitting the pk gives the flexibility for such solutions.